tests/typing/typing_progressbar.py

```diff
@@ -0,0 +1,22 @@
+from typing_extensions import assert_type
+
+from click import progressbar
+from click._termui_impl import ProgressBar
+
+
+def test_length_is_int() -> None:
+    with progressbar(length=5) as bar:
+        assert_type(bar, ProgressBar[int])
+        for i in bar:
+            assert_type(i, int)
+
+
+def it() -> tuple[str, ...]:
+    return ("hello", "world")
```

Comment:
this separate function is entirely so pyright doesn't inference as `Literal[("hello", "world")]`